app/modules/fileuploads/resources.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=bad-continuation
+"""
+RESTful API FileUploads resources
+--------------------------
+"""
+
+import logging
+
+from flask import send_file
+from flask_login import current_user  # NOQA
+from flask_restx_patched import Resource
+from flask_restx_patched._http import HTTPStatus
+
+from app.extensions import db
+from app.extensions.api import Namespace
+from app.extensions.api.parameters import PaginationParameters
+from app.modules.users import permissions
+from app.modules.users.permissions.types import AccessOperation
+
+
+from . import parameters, schemas
+from .models import FileUpload
+
+
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+api = Namespace('fileuploads', description='FileUploads')  # pylint: disable=invalid-name
+
+
+@api.route('/')
+@api.login_required(oauth_scopes=['fileuploads:read'])
+class FileUploads(Resource):
+    """
+    Manipulations with FileUploads.
+    """
+
+    @api.permission_required(
+        permissions.ModuleAccessPermission,
+        kwargs_on_request=lambda kwargs: {
+            'module': FileUpload,
```

Comment:
I suspect you're going to need to add FileUpload to the permitted modules, would be interesting to know what users we expect to allow, all I guess

Reply:
oh yeah sorry -- i had messaged karen and jp about this pr directly; i very much breezed thru a whole ton of things, including just passing on permissions for now.  mostly wanna get karens feedback of the idea in general; then dig into this stuff exactly!